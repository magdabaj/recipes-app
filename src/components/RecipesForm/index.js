/**
 *
 * RecipesForm
 *
 */

// todo return parseInt value in route path selectors

import React, { useState, useEffect } from 'react';
import '../../containers/RecipesFormContainer/index.css';
import RecipeTextInput from '../RecipeTextInput';
import tagTypes from '../../utils/tagTypes';
import Select from '../Select';
import Button from '../Button';
import {Redirect} from "react-router";
import fetchStates from "../../utils/fetchStates";
import {toast} from "react-toastify";

const RecipesForm = ({ status, ...props }) => {
    const [cancel, setCancel] = useState(false)
    const [recipe, setRecipe] = useState({
        title: '',
        website: '',
        url: '',
        image: '',
        tagId: null,
    });
    const [errors, setErrors] = useState({});

    useEffect(() => {
        if (props.recipeId) {
            const _recipe = props.recipes.find(recipe => recipe.id === parseInt(props.recipeId));
            setRecipe(_recipe);
        }
    }, []);

    const handleChange = event => {
        const { name, value } = event.target;

        setRecipe(prevRecipe => ({
            ...prevRecipe,
            [name]: value,
        }));
    };

    const formIsValid = () => {
        const { title, website, url, image, tagId } = recipe;
        const { userId } = props;
        const errors = {};

        if (!title) errors.title = 'Title is required';
        if (!website) errors.website = 'Website is required';
        if (!url) errors.url = 'Url is required';
        if (!image) errors.image = 'Image url is required';
        if (!userId) errors.authorization = 'You must login first';
        if (!tagId) errors.category = 'RecipeWrapper category is required';

        setErrors(errors);

        return Object.keys(errors).length === 0;
    };

    function handleSave(event) {
        event.preventDefault();
        if (!formIsValid()) return;
        try {
            props.addRecipe({ recipe: recipe, userId: parseInt(props.userId) });
        } catch (error) {
            setErrors({ handleSave: error.message });
        }
    }

    return (
        <div className={'recipes-form-container'}>
            <form className={'form'} onSubmit={handleSave}>
                <RecipeTextInput
                    name={'title'}
                    type={'text'}
                    value={recipe.title}
                    label={'Tytuł'}
                    placeholder={'Wpisz tytuł przepisu'}
                    onChange={handleChange}
                    error={errors.title}
                />
                <RecipeTextInput
                    name={'website'}
                    type={'text'}
                    value={recipe.website}
                    label={'Strona'}
                    placeholder={'Wpisz stronę źródłową'}
                    onChange={handleChange}
                    error={errors.website}
                />
                <RecipeTextInput
                    name={'url'}
                    type={'text'}
                    value={recipe.url}
                    label={'Url'}
                    placeholder={'Wpisz adres przepisu'}
                    onChange={handleChange}
                    error={errors.url}
                />
                <RecipeTextInput
                    name={'image'}
                    type={'text'}
                    value={recipe.image}
                    label={'Url zdjęcia'}
                    placeholder={'Wpisz url zdjęcia'}
                    onChange={handleChange}
                    error={errors.image}
                />
                <div className={'select-container'}>
                    {tagTypes.map(tagType => (
                        <Select tags={props.tags} tagType={tagType} onChange={handleChange} />
                    ))}
                    {errors.category && <div className={'error-message'}>{errors.category}</div>}
                </div>
                <div className={'buttons-container'}>
                    <Button
                        type={'submit'}
                        onSubmit={handleSave}
                    >
                        {recipe.id ? 'Edytuj' : 'Zapisz'} Przepis
                    </Button>
                    <Button
                       type={'button'}
                       onClick={() => setCancel(true)}
                    >
                        Cofnij
                    </Button>
                    {cancel && <Redirect to={'/user-recipes'}/>}
                </div>
            </form>
            {status === fetchStates.success ?
                (
                    // toast.success("Recipe added successfully"),
                    <Redirect to={'/user-recipes'}/>
                )
                : null}
        </div>
    );
};

RecipesForm.propTypes = {};

export default RecipesForm;
