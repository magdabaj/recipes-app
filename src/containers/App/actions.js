import {LOAD_RECIPES_ERROR, LOAD_TAGS, LOAD_TAGS_ERROR, LOAD_TAGS_SUCCESS} from "./constants";

// export const loadRecipes = page => ({
//     type: LOAD_RECIPES,
//     page,
// });
//
// export const loadRecipesSuccess = response => ({
//     type: LOAD_RECIPES_SUCCESS,
//     response,
// });

export const loadRecipesError = error => ({
    type: LOAD_RECIPES_ERROR,
    error,
});

export const loadTags = () => ({
    type: LOAD_TAGS,
});

export const loadTagsSuccess = tags => ({
    type: LOAD_TAGS_SUCCESS,
    tags,
});

export const loadTagsError = error => ({
    type: LOAD_TAGS_ERROR,
    error,
});