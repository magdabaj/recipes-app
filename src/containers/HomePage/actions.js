/*
 *
 * RecipesHomePage actions
 *
 */

import {
    CHANGE_PAGE,
    CHANGE_PAGE_ERROR,
    CHANGE_PAGE_SUCCESS,
    LOAD_RECIPES,
    LOAD_RECIPES_BY_TAG,
    LOAD_RECIPES_BY_TAG_ERROR,
    LOAD_RECIPES_BY_TAG_SUCCESS,
    LOAD_RECIPES_ERROR,
    LOAD_RECIPES_SUCCESS,
    LOAD_TAGS,
    LOAD_TAGS_ERROR,
    LOAD_TAGS_SUCCESS,
} from './constants';

export const loadRecipes = page => ({
    type: LOAD_RECIPES,
    page,
});

export const loadRecipesSuccess = response => ({
    type: LOAD_RECIPES_SUCCESS,
    response,
});

export const loadRecipesError = error => ({
    type: LOAD_RECIPES_ERROR,
    error,
});

export const loadRecipesByTag = (tagId, page) => ({
    type: LOAD_RECIPES_BY_TAG,
    tagId,
    page,
});

export const loadRecipesByTagSuccess = response => ({
    type: LOAD_RECIPES_BY_TAG_SUCCESS,
    response,
});

export const loadRecipesByTagError = error => ({
    type: LOAD_RECIPES_BY_TAG_ERROR,
    error,
});

// export const changePage = page => ({
//   type: CHANGE_PAGE,
//   page,
// });
//
// export const changePageSuccess = response => ({
//   type: CHANGE_PAGE_SUCCESS,
//   response,
// });
//
// export const changePageError = error => ({
//   type: CHANGE_PAGE_ERROR,
//   error,
// })
//
