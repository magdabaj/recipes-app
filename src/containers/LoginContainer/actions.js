/*
 *
 * LoginFormContainer actions
 *
 */

import {
    AUTHENTICATE,
    AUTHENTICATE_ERROR,
    AUTHENTICATE_SUCCESS, SET_ERROR,
    SIGN_IN,
    SIGN_IN_ERROR,
    SIGN_IN_SUCCESS,
    SIGN_OUT,
    SIGN_OUT_ERROR,
    SIGN_OUT_SUCCESS,
    SIGN_UP_USER,
    SIGN_UP_USER_ERROR,
    SIGN_UP_USER_SUCCESS
} from './constants'

export const setError = () => ({
    type: SET_ERROR
})

export const signUp = ({ email, password }) => ({
    type: SIGN_UP_USER,
    email,
    password,
});

export const signUpSuccess = (json, email) => ({
    type: SIGN_UP_USER_SUCCESS,
    ...json,
    email,
});

export const signUpError = error => ({
    type: SIGN_UP_USER_ERROR,
    error,
});

export const signOut = () => ({
    type: SIGN_OUT,
});

export const signOutError = error => ({
    type: SIGN_OUT_ERROR,
    error,
});

export const signOutSuccess = json => ({
    type: SIGN_OUT_SUCCESS,
    ...json,
});

export const signIn = ({ email, password }) => ({
    type: SIGN_IN,
    email,
    password,
});

export const signInSuccess = (json, email) => ({
    type: SIGN_IN_SUCCESS,
    ...json,
    email,
});

export const signInError = error => ({
    type: SIGN_IN_ERROR,
    error,
});

export const authenticate = () => ({
    type: AUTHENTICATE,
});

export const authenticateSuccess = json => ({
    type: AUTHENTICATE_SUCCESS,
    ...json,
});

export const authenticateError = error => ({
    type: AUTHENTICATE_ERROR,
    error,
});
