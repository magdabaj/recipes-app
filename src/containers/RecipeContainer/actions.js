/*
 *
 * RecipeContainer actions
 *
 */

import {
    GET_RATINGS_ERROR,
    GET_RATINGS,
    GET_RATINGS_SUCCESS,
    GET_RECIPE,
    GET_RECIPE_ERROR,
    LOAD_TAGS,
    LOAD_TAGS_SUCCESS,
    LOAD_TAGS_ERROR,
    GET_RECIPE_SUCCESS,
    SEND_RATING,
    SEND_RATING_ERROR,
    SEND_RATING_SUCCESS,
    GET_COMMENTS,
    GET_COMMENTS_SUCCESS,
    GET_COMMENTS_ERROR,
    ADD_COMMENT,
    ADD_COMMENT_SUCCESS,
    ADD_COMMENT_ERROR,
    EDIT_COMMENT,
    EDIT_COMMENT_SUCCESS,
    EDIT_COMMENT_ERROR,
    REMOVE_COMMENT,
    REMOVE_COMMENT_SUCCESS,
    REMOVE_COMMENT_ERROR, CLEAR_STATUS, CLEAR_DELETE_COMMENTS_STATUS,
} from './constants';

export const getRecipeRatings = recipeId => ({
    type: GET_RATINGS,
    recipeId,
});

export const getRecipeRatingsSuccess = (message, recipeId, ratingsMean) => ({
    type: GET_RATINGS_SUCCESS,
    message,
    recipeId,
    ratingsMean,
});

export const getRecipeRatingsError = error => ({
    type: GET_RATINGS_ERROR,
    error,
});

export const getRecipe = recipeId => ({
    type: GET_RECIPE,
    recipeId,
});

export const getRecipeSuccess = recipe => ({
    type: GET_RECIPE_SUCCESS,
    recipe,
});

export const getRecipeError = error => ({
    type: GET_RECIPE_ERROR,
    error,
});


export const sendRating = (rate, userId, recipeId) => ({
    type: SEND_RATING,
    rate,
    userId,
    recipeId,
});

export const sendRatingSuccess = (message, recipeId, ratingsMean) => ({
    type: SEND_RATING_SUCCESS,
    message,
    recipeId,
    ratingsMean,
});

export const sendRatingError = error => ({
    type: SEND_RATING_ERROR,
    error,
});

export const getComments = recipeId => ({
    type: GET_COMMENTS,
    recipeId,
});

export const getCommentsSuccess = response => ({
    type: GET_COMMENTS_SUCCESS,
    response,
});

export const getCommentsError = error => ({
    type: GET_COMMENTS_ERROR,
    error,
});

export const addComment = (recipeId, content, userId) => ({
    type: ADD_COMMENT,
    recipeId,
    userId,
    content,
});

export const addCommentSuccess = message => ({
    type: ADD_COMMENT_SUCCESS,
    message,
});

export const addCommentError = error => ({
    type: ADD_COMMENT_ERROR,
    error,
});

export const editComment = (commentId, content, userId) => ({
    type: EDIT_COMMENT,
    commentId,
    userId,
    content,
});

export const editCommentSuccess = message => ({
    type: EDIT_COMMENT_SUCCESS,
    message,
});

export const editCommentError = error => ({
    type: EDIT_COMMENT_ERROR,
    error,
});

export const removeComment = commentId => ({
    type: REMOVE_COMMENT,
    commentId,
});

export const removeCommentSuccess = message => ({
    type: REMOVE_COMMENT_SUCCESS,
    message,
});

export const removeCommentError = error => ({
    type: REMOVE_COMMENT_ERROR,
    error,
});

export const clearStatus = () => ({
    type: CLEAR_STATUS
})

export const clearDeleteCommentsStatus = () => ({
    type: CLEAR_DELETE_COMMENTS_STATUS
})
