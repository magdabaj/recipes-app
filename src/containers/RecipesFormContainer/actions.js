/*
 *
 * RecipesFormContainer actions
 *
 */

import {
    ADD_RECIPE,
    ADD_RECIPE_ERROR,
    ADD_RECIPE_SUCCESS,
    CANCEL, CHANGE_STATUS,
    LOAD_TAGS,
    LOAD_TAGS_ERROR,
    LOAD_TAGS_SUCCESS,
} from './constants';

export const addRecipe = (recipe, userId, tagId) => ({
    type: ADD_RECIPE,
    recipe,
    userId,
    tagId,
});

export const addRecipeSuccess = message => ({
    type: ADD_RECIPE_SUCCESS,
    message,
});

export const addRecipeError = error => ({
    type: ADD_RECIPE_ERROR,
    error,
});

export const changeStatus = () => ({
    type: CHANGE_STATUS,
});

// todo styled components
// todo material ui