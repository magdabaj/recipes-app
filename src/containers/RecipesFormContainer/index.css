
.recipes-form-container {
    max-width: 1170px;
    padding: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
}
.checkbox-container {
    display: block;
    position: relative;
    padding-left: 35px;
    margin-bottom: 12px;
    cursor: pointer;
    font-size: 22px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}

.form {
    background-color: #fff;
    box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
    border-radius: 2px;
    width: 70%;
}

.select-container {
    padding: 2vw;
}

.buttons-container {
    display: flex;
    justify-content: center;
}

/*.label {*/
/*    min-width: 150px;*/
/*    font-size: 14px;*/
/*    font-weight: 500;*/
/*    font-family: Poppins,sans-serif;*/
/*}*/

/*Select component*/
.select {
    text-align: left;
    margin: 1vw;
}
.select-tag-type {
    text-transform: uppercase;
    font-weight: 600;
    width: 100%;
}

.select-tag-type:hover,
.checkbox-label:hover {
    color: #9a1750;
    cursor: pointer;
}

.checkbox-input {
    margin: 5px;
}

[type='checkbox']:not(:checked),
[type='checkbox']:checked {
    position: absolute;
    left: -9999px;
}
[type='checkbox']:not(:checked) + label,
[type='checkbox']:checked + label {
    position: relative;
    padding-left: 1.95em;
    cursor: pointer;
}

/* checkbox aspect */
[type='checkbox']:not(:checked) + label:before,
[type='checkbox']:checked + label:before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 1.25em;
    height: 1.25em;
    border: 2px solid #ccc;
    background: #fff;
    border-radius: 4px;
    box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.1);
}
/* checked mark aspect */
[type='checkbox']:not(:checked) + label:after,
[type='checkbox']:checked + label:after {
    content: '\2713\0020';
    position: absolute;
    top: 0.15em;
    left: 0.22em;
    font-size: 1.3em;
    line-height: 0.8;
    color: #ee4c7c;
    transition: all 0.2s;
}
/* checked mark aspect changes */
[type='checkbox']:not(:checked) + label:after {
    opacity: 0;
    transform: scale(0);
}
[type='checkbox']:checked + label:after {
    opacity: 1;
    transform: scale(1);
}

[type='checkbox']:checked:focus + label:before,
[type='checkbox']:not(:checked):focus + label:before {
    border: 2px dotted #9a1750;
}

/* hover style just for information */
label:hover:before {
    border: 2px solid #9a1750 !important;
}

/*///*/

.recipe-form-button {
    background-color: #fff;
    color: black;
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
    text-transform: uppercase;
    font-size: 1.5vw;
    font-weight: 600;
    padding: 1.5vw 2.5vw;
    margin: 2vw;
    cursor: pointer;
    transition: 0.2s ease-in-out;
    border-radius: 3px;
}

.recipe-form-button:hover {
    background-color: #9a1750;
    color: #fff;
}

.recipe-input-container {
    display: flex;
    align-items: center;
    justify-content: space-around;
    margin: 1.5vw 0;
    width: 100%;
}

@media only screen and (max-width: 768px) {
    .recipe-input-container {
        flex-direction: column;
        justify-content: center;
    }
    .form {
        width: 100%;
    }
    .buttons-container {
        flex-direction: column;
        width: 100%;
    }
    .recipes-form-container {
        margin-top: 20px;
        padding: 20px;
    }
}

@media only screen and (min-width: 768px) {
    .input-data-recipe {
        width: 450px;
    }

    .select-container {
        width: 100%;
    }
}
