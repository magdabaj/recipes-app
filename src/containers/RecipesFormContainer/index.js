/**
 *
 * RecipesFormContainer
 *
 */

// todo logout

import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import makeSelectRecipesFormContainer, {
    makeSelectRecipesAddStatus,
    selectRecipeId,
    selectUserId,
} from './selectors';
import { makeSelectTags } from "../App/selectors";
import RecipesForm from '../../components/RecipesForm';
import { makeSelectUser } from '../LoginContainer/selectors';
import LoginContainer from '../LoginContainer/Loadable';
import { addRecipe, cancelAdding } from './actions';
import { loadTags } from "../App/actions";
import { makeSelectUserRecipes } from '../UserRecipesContainer/selectors';

export function RecipesFormContainer({ ...props }) {
    useEffect(() => {
        if (props.tags.length === 0) props.loadTags();
        else props.loadTags()
    }, []);

    console.log("tags ", props.tags)
    return props.user.loggedIn ? <RecipesForm {...props} /> : <LoginContainer />;
}

RecipesFormContainer.propTypes = {};

const mapStateToProps = createStructuredSelector({
    recipesFormContainer: makeSelectRecipesFormContainer(),
    userId: selectUserId(),
    user: makeSelectUser(),
    status: makeSelectRecipesAddStatus(),
    tags: makeSelectTags(),
    recipeId: selectRecipeId(),
    recipes: makeSelectUserRecipes(),
});

function mapDispatchToProps(dispatch) {
    return {
        addRecipe: (recipe, userId, tagId) => dispatch(addRecipe(recipe, userId, tagId)),
        loadTags: () => dispatch(loadTags()),
    };
}

const withConnect = connect(
    mapStateToProps,
    mapDispatchToProps,
);

export default compose(withConnect)(RecipesFormContainer);
