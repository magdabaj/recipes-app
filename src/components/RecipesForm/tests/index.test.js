import {renderWithRouter} from "../../../utils/testHelpers";
import RecipesForm from "../index";
import {fireEvent, screen, waitFor} from "@testing-library/dom";
import userEvent from "@testing-library/user-event";
import {fakeRecipes} from "../../../utils/testHelpers/fixtures/recipes";
import {addRecipe as addRecipeMocked} from "../../../containers/RecipesFormContainer/actions";
import {fakeTags} from "../../../utils/testHelpers/fixtures/tags";
import React from "react";
import commonTests from "../../../utils/testHelpers/commonTests";
import {axe} from "jest-axe";
import {fakeUser} from "../../../utils/testHelpers/fixtures/user";

jest.mock("../../../containers/RecipesFormContainer/actions")

const renderRecipesForm = () =>
    renderWithRouter(
        <RecipesForm
            recipes={fakeRecipes}
            addRecipe={addRecipeMocked}
            tags={fakeTags}
            userId={fakeUser.id}
        />)

commonTests(renderRecipesForm)

test('renders form without violations', async() => {
    const {container} = renderRecipesForm()
    const results = await axe(container)
    expect(results).toHaveNoViolations()
})

test('fires addRecipes actions when form is correct', async () => {
    renderRecipesForm()
    const titleInput = screen.getByLabelText(/Tytuł/i)
    const websiteInput = screen.getByLabelText(/Strona/i)
    const urlInput = screen.getByLabelText("Url")
    const imageInput = screen.getByLabelText(/Url zdjęcia/i)
    const tagCheckbox = screen.getByTestId('alkohol')

    const saveButton = screen.getByText(/zapisz przepis/i)

    await userEvent.type(titleInput, 'test')
    await userEvent.type(websiteInput, 'test')
    await userEvent.type(urlInput, 'test')
    await userEvent.type(imageInput, 'test')
    // expect(checkbox).not.toBeInTheDocument()

    userEvent.click(tagCheckbox)

    const checkbox = screen.getByText(/rum/i)
    expect(checkbox).toBeInTheDocument()
    fireEvent.click(checkbox, { target: { checked: true } });
    await waitFor(() => userEvent.click(saveButton))

    expect(addRecipeMocked).toHaveBeenCalled()
    expect(addRecipeMocked).toHaveBeenCalledTimes(1)
})

test('shows error message if tag is not chosen',   () => {
    renderRecipesForm()
    const titleInput = screen.getByLabelText(/Tytuł/i)
    const websiteInput = screen.getByLabelText(/Strona/i)
    const urlInput = screen.getByLabelText("Url")
    const imageInput = screen.getByLabelText(/Url zdjęcia/i)
    const tagCheckbox = screen.getByTestId('warzywa, grzyby')

    // expect(errorMessage).not.toBeInTheDocument()

    const saveButton = screen.getByText(/zapisz przepis/i)

    userEvent.type(titleInput, 'test')
    userEvent.type(websiteInput, 'test')
    userEvent.type(urlInput, 'test')
    userEvent.type(imageInput, 'test')
    userEvent.click(tagCheckbox)
    userEvent.click(saveButton)
    const errorMessage = screen.getByText(/kategoria jest wymagana/i)
    expect(errorMessage).toBeInTheDocument()
})

test('calls setCancel function on click', async () => {
    const { history } = renderRecipesForm()

    const cancelButton = screen.getByText(/cofnij/i)
    userEvent.click(cancelButton)

    await waitFor(() => {
        expect(history.location.pathname).toEqual('/user-recipes')
    })

})

test('doesn\'t call the addRecipe function when inputs are empty', () => {
    renderRecipesForm()

    const saveButton = screen.getByText(/zapisz przepis/i)

    userEvent.click(saveButton)

    expect(addRecipeMocked).not.toHaveBeenCalled()
})

test('Displays different texts when editing recipe', () => {
    renderWithRouter(
        <RecipesForm
            recipes={fakeRecipes}
            addRecipe={addRecipeMocked}
            tags={fakeTags}
            userId={fakeUser.id}
            recipeId={'1'}
        />
    )

    const editButton = screen.queryByText(/edytuj przepis/i)
    expect(editButton).toBeInTheDocument()
})