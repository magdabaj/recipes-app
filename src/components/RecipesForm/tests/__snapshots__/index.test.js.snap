// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="recipes-form-container"
>
  <form
    class="form"
  >
    <div
      class="recipe-input-container"
    >
      <label
        class="label"
        for="title"
      >
        Tytuł
      </label>
      <div
        class="input-data input-data-recipe"
      >
        <input
          class="input"
          id="title"
          name="title"
          placeholder="Wpisz tytuł przepisu"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="recipe-input-container"
    >
      <label
        class="label"
        for="website"
      >
        Strona
      </label>
      <div
        class="input-data input-data-recipe"
      >
        <input
          class="input"
          id="website"
          name="website"
          placeholder="Wpisz stronę źródłową"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="recipe-input-container"
    >
      <label
        class="label"
        for="url"
      >
        Url
      </label>
      <div
        class="input-data input-data-recipe"
      >
        <input
          class="input"
          id="url"
          name="url"
          placeholder="Wpisz adres przepisu"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="recipe-input-container"
    >
      <label
        class="label"
        for="image"
      >
        Url zdjęcia
      </label>
      <div
        class="input-data input-data-recipe"
      >
        <input
          class="input"
          id="image"
          name="image"
          placeholder="Wpisz url zdjęcia"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="select-container"
    >
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="alkohol"
          for="alkohol"
        >
          alkohol
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="bakalie"
          for="bakalie"
        >
          bakalie
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="czekolada, znane slodycze"
          for="czekolada, znane slodycze"
        >
          czekolada, znane slodycze
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="dodatki"
          for="dodatki"
        >
          dodatki
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="herbata, kawa"
          for="herbata, kawa"
        >
          herbata, kawa
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="maki, platki, ziarna"
          for="maki, platki, ziarna"
        >
          maki, platki, ziarna
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="masla, kremy, curdy"
          for="masla, kremy, curdy"
        >
          masla, kremy, curdy
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="miody, syropy"
          for="miody, syropy"
        >
          miody, syropy
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="owoce"
          for="owoce"
        >
          owoce
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="przyprawy, ziola"
          for="przyprawy, ziola"
        >
          przyprawy, ziola
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="smietany, jogurty, mleczka"
          for="smietany, jogurty, mleczka"
        >
          smietany, jogurty, mleczka
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="twarogi, serki, sery"
          for="twarogi, serki, sery"
        >
          twarogi, serki, sery
        </label>
      </div>
      <div
        class="select"
      >
        <label
          class="select-tag-type"
          data-testid="warzywa, grzyby"
          for="warzywa, grzyby"
        >
          warzywa, grzyby
        </label>
      </div>
    </div>
    <div
      class="buttons-container"
    >
      <button
        class="Button-sc-1iw2tsm-0 gQWKh"
        type="submit"
      >
        Zapisz
         Przepis
      </button>
      <button
        class="Button-sc-1iw2tsm-0 gQWKh"
        type="button"
      >
        Cofnij
      </button>
    </div>
  </form>
</div>
`;
